### PR TITLE
ci(test): add new pytest coverage comment tool

### LINKED ISSUE #25: Switch to `pytest-coverage-comment`
[pytest-coverage-commentator](https://github.com/coroo/pytest-coverage-commentator) has stopped update for 3+ years and produces tons of deprecation warnings.

Maybe we can switch to [pytest-coverage-comment](https://github.com/MishaKav/pytest-coverage-comment).

### PR DESCRIPTION
**Pull request by emptymalei**
Resolves #25

Changed:
- pytest coverage comment step in actions

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -37,6 +33,11 @@ jobs:
...
-      - name: Comment coverage
-        uses: coroo/pytest-coverage-commentator@v1.0.2
+          poetry run pytest --cache-clear --junitxml=pytest.xml --cov=hamilflow tests/ > pytest-coverage.txt
+      - name: Pytest coverage comment
+        uses: MishaKav/pytest-coverage-comment@v1.1.51
+        with:
+          pytest-coverage-path: ./pytest-coverage.txt
+          junitxml-path: ./pytest.xml
+          create-new-comment: false
+          title: Coverage Report (${{ github.event.pull_request.head.sha || github.sha }}, ${{ matrix.python-version }}, ${{ matrix.os }})
```
Comment (GrumpySapiens) [inline .github/workflows/tests.yaml:43]: I mean to print the poetry version

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -37,6 +33,11 @@ jobs:
...
-          poetry run pytest --cache-clear --cov=hamilflow tests/ > pytest-coverage.txt
-      - name: Comment coverage
-        uses: coroo/pytest-coverage-commentator@v1.0.2
+          poetry run pytest --cache-clear --junitxml=pytest.xml --cov=hamilflow tests/ > pytest-coverage.txt
+      - name: Pytest coverage comment
+        uses: MishaKav/pytest-coverage-comment@v1.1.51
+        with:
+          pytest-coverage-path: ./pytest-coverage.txt
+          junitxml-path: ./pytest.xml
+          create-new-comment: false
```
Comment (GrumpySapiens) [inline .github/workflows/tests.yaml:42]: Default is set to false, wonder if needed to specify the default value here

File: .github/workflows/tests.yaml [hunk 3]
```diff
@@ -37,6 +33,11 @@ jobs:
         run: poetry install --with docs,tutorial
       - name: Build coverage file
         run: |
-          poetry run pytest --cache-clear --cov=hamilflow tests/ > pytest-coverage.txt
-      - name: Comment coverage
-        uses: coroo/pytest-coverage-commentator@v1.0.2
+          poetry run pytest --cache-clear --junitxml=pytest.xml --cov=hamilflow tests/ > pytest-coverage.txt
```
[the remark this answers is not in the document]
Comment (emptymalei) [reply .github/workflows/tests.yaml:36]: yeah, this was a copy paste from an old project.

Fixed in 01272a9

File: .github/workflows/tests.yaml (hunk 2 above)
Comment (emptymalei) [reply .github/workflows/tests.yaml]: 01272a9

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (emptymalei) [reply .github/workflows/tests.yaml:42]: not sure if that is gonna be useful.

File: .github/workflows/tests.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (GrumpySapiens) [reply .github/workflows/tests.yaml:36]: I just configured my laptop to have 1.3.2 T_T, anyway better to have the latest

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (GrumpySapiens) [reply .github/workflows/tests.yaml:42]: As long as it's synced with the lock file, I am fine

File: .github/workflows/tests.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (GrumpySapiens) [reply .github/workflows/tests.yaml:36]: Looks good to me, not sure how to resolve conversations tho...

**Comment by GrumpySapiens** [comment]
👍

**Review by GrumpySapiens — APPROVED** [review]
Looks good to me

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (emptymalei) [reply .github/workflows/tests.yaml:42]: `poetry` is sort of global on each of our machine and its version in Actions is set manually.
I guess we have to agree on some conventions on this problem. 
For now, I would say, we don't deal with it unless we run into problems?

Comment (GrumpySapiens) [reply .github/workflows/tests.yaml:42]: Fine by me. Poetry should be back compatible enough (hopefully, being such a high profile tool), at least for minor versions.

**Pull request merged**